src/curve.rs

```diff
@@ -189,6 +189,18 @@ impl CompressedMontgomeryU {
         // y = (u - 1) / (u + 1)
 
         let u = FieldElement::from_bytes(&self.0);
+
+        // If u = -1, then v^2 = u*(u^2+486662*u+1) = 486660.
+        // But 486660 is nonsquare mod p, so this is not a curve point.
+        //
+        // XXX what does Signal do here?
```

Comment:
Signal uses Montgomery `u` to derive Edwards `y` (by `y = (u-1)/(u+1)`, `u` to derive `v` (by `v = sqrt(u(u² + Au + 1))`), `(u, v)` to derive Edwards `x` (by `x = (u/v) * sqrt(-(A+2))`), It also expects to have `ge_montx_to_p3()` be passed a mask indicating the sign, and it conditionally negates such that the resulting Edwards `x` is always positive.

Maybe we should just copy that behaviour? If for no other reason than it has one less inversion.